src/Castle.Windsor/MicroKernel/Releasers/LifecycledComponentsReleasePolicy.cs

```diff
@@ -84,33 +84,43 @@ private Burden[] TrackedObjects
 		{
 			get
 			{
-				using (var holder = @lock.ForReading(false))
+				@lock.EnterReadLock();
```

Comment:
Not sure if you want to preserve exact semantics here (probably not, since you're removing the below TODO)... but if you do, `TryEnterReadLock` would be the right method to call instead.